server/src/test/java/org/elasticsearch/index/shard/PrimaryReplicaSyncerTests.java

```diff
@@ -86,9 +94,19 @@ public void testSyncerSendsOffCorrectDocuments() throws Exception {
         syncer.resync(shard, fut);
         fut.get();
 
-        if (syncNeeded) {
```

Comment:
can you just change the definition of sync needed? it's weird to have it false while sync is still going out.